mvpa2/misc/surfing/queryengine.py

```diff
@@ -153,6 +155,9 @@ def train(self, ds):
         for feature_id, vertex_id in enumerate(vertex_ids):
             v2f[vertex_id].append(feature_id)
 
+        # store ids only for those vertices for which we have neighbors
+        self._ids = [k for k in self._vertex2feature_map.keys()
+                     if len(self._vertex2feature_map[k])]
```

Comment:
Because `Searchlight` loops through `qe.ids`. If the id is not present in the dataset, then it doesn't have any neighbor in `SurfaceQueryEngine`. However, `Searchlight` will still use it as a center.

I guess this comes from an interpretation of how `Searchlight` should work, and I might have missed it. I expect that if I pass a dataset without a particular node_index `i`, then node index `i` shouldn't be used as a center for searchlight. In the current version, it will be used as a center anyway (because it exists on the surface), and will take as neighbors not the immediate nodes, but those that fall within the radius.

One can specify `roi_ids` to avoid this behavior, but to me it makes more sense that if I mask a dataset, then only the nodes in the mask dataset should participate in the searchlight.